Sources/Components/StepSlider/StepSlider.swift

```diff
@@ -214,3 +252,68 @@ public final class StepSlider: UISlider {
         }
     }
 }
+
+class ThumbImageFactory {
+    let accentColor: UIColor
+    let diameter: CGFloat
+    let centerRadius: CGFloat
+    let highlightedCenterRadius: CGFloat
+    let bounds: CGRect
+    let shadowSize: CGFloat = 5
+
+    init(diameter: CGFloat, centerRadius: CGFloat, highlightedCenterRadius: CGFloat, accentColor: UIColor) {
+        self.diameter = diameter
+        self.centerRadius = centerRadius
+        self.highlightedCenterRadius = highlightedCenterRadius
+        self.accentColor = accentColor
+        self.bounds = CGRect(origin: .zero, size: CGSize(width: diameter, height: diameter))
+    }
+
+    var image: UIImage? {
+        if _image == nil {
+            _image = computeImage(effectiveRadius: centerRadius)
+        }
+        return _image
+    }
+
+    var highligtedImage: UIImage? {
+        if _highligtedImage == nil {
+            _highligtedImage = computeImage(effectiveRadius: highlightedCenterRadius)
+        }
+        return _highligtedImage
+    }
+
+    private var _image: UIImage?
+    private var _highligtedImage: UIImage?
+
+    private func computeImage(effectiveRadius: CGFloat) -> UIImage {
+        let sizeWithShadow = CGSize(width: diameter + shadowSize * 2, height: diameter + shadowSize * 2)
+        let renderer = UIGraphicsImageRenderer(size: sizeWithShadow)
+        return renderer.image { (context) in
+            context.cgContext.saveGState()
+            context.cgContext.setShadow(
+                offset: CGSize(width: 0, height: 1),
+                blur: shadowSize,
+                color: UIColor(white: 0.0, alpha: 0.3).cgColor
+            )
+
+            UIColor.white.setFill()
+            context.cgContext.fillEllipse(in: bounds.applying(.init(translationX: shadowSize, y: shadowSize)))
+            context.cgContext.restoreGState()
+
+            self.accentColor.setFill()
+            let center = bounds
```

Comment:
Shouldn't this always be the center of the rect or I'm missing something?

Reply:
maybe i can improve the name.
the intention is _the origin of the rect where the concentric circle will be_
if that origin is the center of the outer rect itself then the circle would not be centered